packages/open-truss/src/signals/index.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from 'zod'
+import { signal, type Signal } from '@preact/signals-react'
+export { effect, computed } from '@preact/signals-react'
+export type SignalsZodType = z.ZodDefault<z.ZodType<Signal<any>>>
+export type Signals = Record<string, Signal<any>>
+export type SignalTypes = Record<string, SignalsZodType>
+export const SIGNALS: SignalTypes = {}
```

Comment:
Might be nice to have a comment here about how this is used (I just had to look it up and it's not the easiest grep).

> We keep a global store of all SignalTypes to map the string of a signal's declared type e.g. `number[]` to the actual preact signal of that type.

Reply:
Good idea!